subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/ConfigurationCacheBuildOperations.kt

```diff
@@ -24,17 +26,36 @@ import org.gradle.internal.operations.CallableBuildOperation
 
 internal
 fun <T : Any> BuildOperationExecutor.withLoadOperation(block: () -> T) =
-    withOperation("Load configuration cache state", block)
+    withOperation("Load configuration cache state", block, loadDetails, loadResult)
 
 
 internal
 fun BuildOperationExecutor.withStoreOperation(cacheKey: String, block: () -> Unit) =
-    withOperation("Store configuration cache state $cacheKey", block)
+    withOperation("Store configuration cache state $cacheKey", block, storeDetails, storeResult)
 
 
 private
-fun <T : Any> BuildOperationExecutor.withOperation(displayName: String, block: () -> T): T =
+val loadDetails = object : ConfigurationCacheLoadBuildOperationType.Details {}
```

Comment:
💄 `loadDetails`, `loadResult`, `storeDetails` and `storeResult` could be defined as named objects instead of vals, `object LoadResult : ...`, etc.

Reply:
done